src/crimson/os/seastore/segment_cleaner.cc

```diff
@@ -435,6 +464,60 @@ SegmentCleaner::gc_reclaim_space_ret SegmentCleaner::gc_reclaim_space()
   });
 }
 
+SegmentCleaner::scan_extents_ret SegmentCleaner::scan_nonfull_segment(
+  const segment_header_t& header,
+  init_segments_ret_bare& segment_set,
+  segment_id_t segment_id)
+{
+  if (header.out_of_line) {
+    logger().info(
+      "ExtentReader::init_segments: out-of-line segment {}",
+      segment_id);
+    return seastar::do_with(
+      scan_valid_records_cursor({
+	segments[segment_id].journal_segment_seq,
+	paddr_t::make_seg_paddr(segment_id, 0)}),
+      [this, segment_id](auto& cursor) {
+      return scanner->scan_extents(
```

Comment:
I think we can instead scan record header if both the commit-time and commit-type are moved to the header.

Reply:
I think we have to scan_extents either way, because we have to set `last_modified`/`last_rewritten` for each extent.